drivers/storage/portworx/components_test.go

```diff
@@ -9737,6 +9738,46 @@ func TestDisablePodSecurityPolicies(t *testing.T) {
 	require.Empty(t, len(policies.Items))
 }
 
+func TestTelemetryEnable(t *testing.T) {
+	coreops.SetInstance(coreops.New(fakek8sclient.NewSimpleClientset()))
+	reregisterComponents()
+	k8sClient := testutil.FakeK8sClient()
+	driver := portworx{}
+	driver.Init(k8sClient, runtime.NewScheme(), record.NewFakeRecorder(0))
+
+	cluster := &corev1.StorageCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "px-cluster",
+			Namespace: "kube-test",
+			Annotations: map[string]string{
+				pxutil.AnnotationTelemetryArcusLocation: "internal",
+			},
+		},
+		Spec: corev1.StorageClusterSpec{
+			Monitoring: &corev1.MonitoringSpec{
+				Telemetry: &corev1.TelemetrySpec{
+					Enabled: true,
+					Image:   "portworx/px-telemetry:2.1.2",
+				},
+			},
+		},
+	}
+
+	err := driver.PreInstall(cluster)
+	require.NoError(t, err)
+
+	// CCM config map
+	expectedConfigMap := testutil.GetExpectedConfigMap(t, "ccmconfig.yaml")
+	telemetryConfigMap := &v1.ConfigMap{}
+	err = testutil.Get(k8sClient, telemetryConfigMap, component.TelemetryConfigMapName, cluster.Namespace)
+	require.NoError(t, err)
+	require.Equal(t, expectedConfigMap.Name, telemetryConfigMap.Name)
+	require.Equal(t, expectedConfigMap.Namespace, telemetryConfigMap.Namespace)
+	require.Len(t, telemetryConfigMap.OwnerReferences, 1)
+	require.Equal(t, cluster.Name, telemetryConfigMap.OwnerReferences[0].Name)
+	require.Equal(t, expectedConfigMap.Data, telemetryConfigMap.Data)
+}
```

Comment:
Can we add tests for
- when arcus location annotation is not passed?
- when telemetry is disabled, the configmap should go away (checkout [other component disable tests](https://github.com/libopenstorage/operator/blob/master/drivers/storage/portworx/components_test.go#L8299))

Reply:
added